flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestEndpointITCase.java

```diff
@@ -86,46 +98,97 @@ public void testEndpoints() throws Exception {
 			mockGatewayRetriever,
 			RpcUtils.INF_TIMEOUT);
 
-		RestServerEndpoint serverEndpoint = new TestRestServerEndpoint(serverConfig, testHandler);
-		RestClient clientEndpoint = new TestRestClient(clientConfig);
+		serverEndpoint = new TestRestServerEndpoint(serverConfig, testHandler);
+		clientEndpoint = new TestRestClient(clientConfig);
 
-		try {
-			serverEndpoint.start();
-
-			TestParameters parameters = new TestParameters();
-			parameters.jobIDPathParameter.resolve(PATH_JOB_ID);
-			parameters.jobIDQueryParameter.resolve(Collections.singletonList(QUERY_JOB_ID));
-
-			// send first request and wait until the handler blocks
-			CompletableFuture<TestResponse> response1;
-			synchronized (TestHandler.LOCK) {
-				response1 = clientEndpoint.sendRequest(
-					serverConfig.getEndpointBindAddress(),
-					serverConfig.getEndpointBindPort(),
-					new TestHeaders(),
-					parameters,
-					new TestRequest(1));
-				TestHandler.LOCK.wait();
-			}
+		serverEndpoint.start();
+	}
+
+	@After
+	public void teardown() {
+		if (clientEndpoint != null) {
+			clientEndpoint.shutdown(timeout);
+			clientEndpoint = null;
+		}
 
-			// send second request and verify response
-			CompletableFuture<TestResponse> response2 = clientEndpoint.sendRequest(
-				serverConfig.getEndpointBindAddress(),
-				serverConfig.getEndpointBindPort(),
+		if (serverEndpoint != null) {
+			serverEndpoint.shutdown(timeout);
+			serverEndpoint = null;
+		}
+	}
+
+	@Test
+	public void testEndpoints() throws Exception {
```

Comment:
rename to "testRequestInterleaving", since we modify the line anyway

Reply:
Will do